src/main/java/de/tum/in/www1/artemis/domain/plagiarism/text/TextPlagiarismResult.java

```diff
@@ -29,7 +33,14 @@ public void convertJPlagResult(JPlagResult result, Exercise exercise) {
             this.comparisons.add(comparison);
         }
         this.duration = result.getDuration();
-        this.setSimilarityDistribution(result.getSimilarityDistribution());
+
+        // Convert JPlag Similarity Distribution from int[100] to int[10]
+        int[] tenthPercentileSimilarityDistribution = new int[REDUCED_SIZE];
+        for (int i = 0; i < ORIGINAL_SIZE; i++) {
+            tenthPercentileSimilarityDistribution[Math.floorDiv(i, REDUCED_SIZE)] += result.getSimilarityDistribution()[i];
```

Comment:
```suggestion
            tenthPercentileSimilarityDistribution[i / REDUCED_SIZE] += result.getSimilarityDistribution()[i];
```
Normal integer division already floors the result